### PR TITLE
Zaak statuses endpoint for BFF

### PR DESCRIPTION
**Pull request by sergei-maertens**
Sample response:

```json
[
    {
        "url": "https://openzaak.utrechtproeftuin.nl/zaken/api/v1/statussen/4d962ace-3483-4187-a859-e41d7b757613",
        "datumStatusGezet": "2020-12-18T13:44:00.405552Z",
        "statustoelichting": "",
        "statustype": {
            "url": "https://openzaak.utrechtproeftuin.nl/catalogi/api/v1/statustypen/e8140a06-2a54-4c2c-b5db-6e4c4e20b9aa",
            "omschrijving": "Concept rapport ter accordering voorgelegd",
            "omschrijvingGeneriek": "Rapport concept ter accordering",
            "statustekst": "",
            "volgnummer": 2,
            "isEindstatus": false
        }
    },
    {
        "url": "https://openzaak.utrechtproeftuin.nl/zaken/api/v1/statussen/c6b21d05-2ed6-4a67-8318-9e41e71d5238",
        "datumStatusGezet": "2020-12-18T13:43:38.488494Z",
        "statustoelichting": "",
        "statustype": {
            "url": "https://openzaak.utrechtproeftuin.nl/catalogi/api/v1/statustypen/2c11f327-6755-41c7-bc6b-882fa08fb0f1",
            "omschrijving": "Ontvangen",
            "omschrijvingGeneriek": "Ontvangen",
            "statustekst": "",
            "volgnummer": 1,
            "isEindstatus": false
        }
    }
]
```

Ordered by most recent first.

### CODE REVIEW

File: backend/src/zac/core/api/views.py [hunk 1]
```diff
@@ -210,15 +217,33 @@ def get(self, request: Request) -> Response:
...
             zaak = find_zaak(**self.kwargs)
         except ObjectDoesNotExist:
             raise Http404("No zaak matches the given query.")
-        self.check_object_permissions(request, zaak)
+        self.check_object_permissions(self.request, zaak)
+        return zaak
+
+
+class ZaakDetailView(GetZaakMixin, views.APIView):
+    authentication_classes = (authentication.SessionAuthentication,)
```
Comment (damm89) [inline backend/src/zac/core/api/views.py:231]: https://github.com/GemeenteUtrecht/zaakafhandelcomponent/blob/0285d92edd446209e35d3211f10f471e5814feea/backend/src/zac/conf/includes/base.py#L416

This already handled by default authentication?

File: backend/src/zac/core/api/views.py [hunk 2]
```diff
@@ -210,15 +217,33 @@ def get(self, request: Request) -> Response:
...
+    permission_classes = (permissions.IsAuthenticated & CanReadZaken,)
+
+    def get(self, request, *args, **kwargs):
+        zaak = self.get_object()
         serializer = ZaakDetailSerializer(instance=zaak)
         return Response(serializer.data)
+
+
+class ZaakStatusesView(GetZaakMixin, views.APIView):
+    authentication_classes = (authentication.SessionAuthentication,)
```
Comment (damm89) [inline backend/src/zac/core/api/views.py:241]: https://github.com/GemeenteUtrecht/zaakafhandelcomponent/blob/0285d92edd446209e35d3211f10f471e5814feea/backend/src/zac/conf/includes/base.py#L416

Comment (damm89) [reply backend/src/zac/core/api/views.py:241]: See comment above.

**Review by damm89 — APPROVED** [review]
Looks good - one comment.

**Comment by kelvincy** [comment]
Status per action looks good for the timeline!

Is this endpoint also designed for the following?
- Status bar with the progress percentage
- Deadline date?

See image below for clarification

Screenshot 2020-12-24 at 15 44 16

**Review by annashamray — APPROVED** [review]
Looks good!

File: backend/src/zac/core/api/views.py (hunk 1 above)
Comment (sergei-maertens) [reply backend/src/zac/core/api/views.py:231]: yeah but I want it to be handled _only_ by Session auth, not by Token auth

**Comment by sergei-maertens** [comment]
> Is this endpoint also designed for the following?

that's data that is derived from the zaak detail endpoint and zaaktype - see https://github.com/GemeenteUtrecht/zaakafhandelcomponent/blob/develop/backend/src/zgw/models/zrc.py#L17 and https://github.com/GemeenteUtrecht/zaakafhandelcomponent/blob/develop/backend/src/zgw/models/zrc.py#L17

**Comment by sergei-maertens** [comment]
@kelvincy the above referenced code has been added to the zaak detail endpoint - `deadline` and `deadlineProgress` keys are now present.

**Review by kelvincy — APPROVED** [review]

**Pull request merged**